vendor/blockchain_contracts/src/rfc003/timestamp.rs

```diff
@@ -20,6 +27,29 @@ impl Timestamp {
     pub fn plus(self, seconds: u32) -> Self {
         Self(self.0.checked_add(seconds).unwrap_or(std::u32::MAX))
     }
+
+    pub fn to_vec(self, len: usize) -> Result<Vec<u8>, ToVecError> {
+        let mut vec = Vec::with_capacity(len);
+        vec.write_u32::<BigEndian>(self.0).map_err(ToVecError::Io)?;
```

Comment:
As mentioned above this should be implemented as a trait `EncodeEthereum` and this whole function can just be:

```rust
pub fn to_vec(&self) -> Vec<u8> {
  let mut wtr = vec![];
  wtr.write_u32::<BigEndian>(self.0).expect("writing to Vec can't fail");
  return wtr;
}
```